ipn/ipnlocal/node_backend.go

```diff
@@ -378,20 +383,54 @@ func (nb *nodeBackend) PeerIsReachable(ctx context.Context, p tailcfg.NodeView)
 	nb.mu.Unlock()
 
 	if self := nm.SelfNode; self.Valid() && self.ID() == p.ID() {
-		// This node can always reach itself.
+		// Nodes can always reach themselves.
 		return true
 	}
-	return nb.peerIsReachable(ctx, p)
+	res, err := nb.peerIsReachable(ctx, p)
+	if err != nil {
+		nb.logf("%s", err)
+	}
+	return res
 }
 
-func (nb *nodeBackend) peerIsReachable(ctx context.Context, p tailcfg.NodeView) bool {
+func (nb *nodeBackend) peerIsReachable(ctx context.Context, p tailcfg.NodeView) (bool, error) {
```

Comment:
document what (false, nil) means and how that differs from (false, context expired) ?